sycl/test/check_device_code/cuda/ldg.cpp

```diff
@@ -9,279 +10,509 @@ using namespace sycl;
 using namespace sycl::ext::oneapi::experimental::cuda;
 using namespace sycl::ext::oneapi::experimental;
 
-int main() {
+// CHECK-OPAQUE-LABEL: define weak_odr dso_local %"class.sycl::_V1::detail::half_impl::half" @_ZN4sycl3_V13ext6oneapi12experimental4cuda3ldgINS0_6detail9half_impl4halfEEENSt9enable_ifIXsr4sycl6detail12is_containedIT_NS6_5boost4mp117mp_listIJNS0_3vecIcLi2EEENSE_IcLi3EEENSE_IcLi4EEENSE_IaLi2EEENSE_IaLi3EEENSE_IaLi4EEENSE_IsLi2EEENSE_IsLi3EEENSE_IsLi4EEENSE_IiLi2EEENSE_IiLi3EEENSE_IiLi4EEENSE_IlLi2EEENSE_IlLi3EEENSE_IlLi4EEENSE_IxLi2EEENSE_IxLi3EEENSE_IxLi4EEENSE_IhLi2EEENSE_IhLi3EEENSE_IhLi4EEENSE_ItLi2EEENSE_ItLi3EEENSE_ItLi4EEENSE_IjLi2EEENSE_IjLi3EEENSE_IjLi4EEENSE_ImLi2EEENSE_ImLi3EEENSE_ImLi4EEENSE_IyLi2EEENSE_IyLi3EEENSE_IyLi4EEENSE_IS8_Li2EEENSE_IS8_Li3EEENSE_IS8_Li4EEENSE_IfLi2EEENSE_IfLi3EEENSE_IfLi4EEENSE_IdLi2EEENSE_IdLi3EEENSE_IdLi4EEEfdS8_NS2_8bfloat16EcasilxhtjmySt4byteEEEEE5valueESA_E4typeEPKSA_(
```

Comment:
I think that all this checking is probably overkill, the purpose of the test is really to only check these parts:
```
tail call half @llvm.nvvm.ldg.global.f.f16.p0
```

The other stuff can (and sometimes does) change, but is orthogonal to the purposes of these tests.

Could you update it to be more consistent with how the checks were previously, which were more focused on checking the correct llvm builtins are being generated?

Reply:
That makes sense, I've gone ahead and removed all the other tests.

Reply:
Thanks, it looks good. Most of the files are missing EOF. If you can fix that then we can run the testing on this.